### DISCUSSION TITLE
How to show image in dify?

### DISCUSSION THREAD

**Discussion by silinsi** [langgenius/dify#10441] [Help] [answered]
### Self Checks

- [X] I have searched for existing issues [search for existing issues](https://github.com/langgenius/dify/issues), including closed ones.
- [X] I confirm that I am using English to submit this report (我已阅读并同意 [Language Policy](https://github.com/langgenius/dify/issues/1542)).
- [X] [FOR CHINESE USERS] 请务必使用英文提交 Issue，否则会被关闭。谢谢！:）
- [X] Please do not modify this template :) and fill in all the required fields.

### 1. Is this request related to a challenge you're experiencing? Tell me about your story.

I use comfyui tool to generate the image, however, link the output file to end node make no sense. I could only get the output image by using the url to download, or open a page on browser.
Is there any good method to show the image on dify? I know llm module can support image upload and display, what about other nodes?

### 2. Additional context or comments

![image](https://github.com/user-attachments/assets/45b1ddba-f01f-4cb0-a32e-4a49312ddda7)

**Comment by silinsi** [comment, marked as answer]
Alright, i find a way. i release the pipeline i made as a tool. Then use an agent to call the tool, the llm can show the image in the interface!

**Reply by LinsenHou** [reply]
Dear silinsi,
   I meet the same problem about how to show image in the conversion, can you share more info about the solution ? Thanks a lot.

**Reply by silinsi** [reply]
> Dear silinsi, I meet the same problem about how to show image in the conversion, can you share more info about the solution ? Thanks a lot.

If you get the "url" similar to the image ahead, you can release the workflow as a tool("publish"->"workflow as tool""configure").
Then create an agent, and add the tool, with proper prompt, the agent will use the tool, and show the image in the conversion.
工作流右上角发布为工具，配置好，之后新建agent，在工具里加上这个workflow，写个提示词，然后agent就会调用这个工具，结果的图片能在对话框里直接显示
![QQ图片20241113172722](https://github.com/user-attachments/assets/8f8e29aa-54be-4e12-be1a-735f598dbf5a)

**Reply by LinsenHou** [reply]
嗯，我这边是拼好了URL，然后对话中展示的是URL，就像你描述的，需要点开才能预览照片，而不是在对话中直接展示。
![image](https://github.com/user-attachments/assets/7098d429-f34c-4503-aafc-5a79876a56c7)
我试试你说的配置成工具的方法。

**Reply by LinsenHou** [reply]
你好，之前没用过工作流，还得请教下，在工作流里编排的时候没有sys.query这个变量呢，这咋使用？
![image](https://github.com/user-attachments/assets/cc6246bd-8319-464a-96ec-2f111347419d)

**Reply by silinsi** [reply]
> 你好，之前没用过工作流，还得请教下，在工作流里编排的时候没有sys.query这个变量呢，这咋使用？ ![image](https://private-user-images.githubusercontent.com/9674322/385693575-cc6246bd-8319-464a-96ec-2f111347419d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzE2NTc4NDAsIm5iZiI6MTczMTY1NzU0MCwicGF0aCI6Ii85Njc0MzIyLzM4NTY5MzU3NS1jYzYyNDZiZC04MzE5LTQ2NGEtOTZlYy0yZjExMTM0NzQxOWQucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTExNSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDExMTVUMDc1OTAwWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9YzcxYjJlZTY2ZjVmM2NhNjRhMmY5OTQwNDc1MDMxZGZiNWYyNzgzMjBlNDY4OTRjNThhMDZiZjhmODMyZjJhYiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.DQxoojSRrMeccMW8yMsovqCR_QdlOR4mAXN-IoV8fOA)

我没用过sys.query这个变量，我的做法是在流水线里，将TXT2IMG工具连接到结束节点，在结束节点输出files，不用手动拼接URL
![image](https://github.com/user-attachments/assets/c6b27054-1e29-4c44-aa71-bd7dfd830945)

**Reply by LinsenHou** [reply]
收到，明白啦，感谢~
